### ISSUE TITLE
state file.copy for directories does not set ownership recursively

### ISSUE THREAD

**Issue by systemtrap** [saltstack/salt#39445]
### state file.copy for directories does not set ownership recursively

The commit aead8a9ae447ea20959b7a624186b7d54f06e9a2 seems to incomplete.

### Setup
> `mkdir -p /src/sub1`
> touch /src/sub1/txt

```
test_state:
  file.copy:
    - name: /src
    - source: /dst
    - makedirs: true
    - user: dummy
    - group: dummy
    - subdir: True
```

### Steps to Reproduce Issue
1. execute state
2. check permissions of /dst/sub

The ownership of /dst is dummy:dummy. The ownership of /dst/sub is something like root:root.

### Versions Report

```
Salt Version:
           Salt: 2016.11.2

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 1.4.1
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.3
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.20.2
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.6.6 (r266:84292, May 22 2015, 08:34:51)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: redhat 6.7 Santiago
        machine: x86_64
        release: 2.6.32-573.35.2.el6.x86_64
         system: Linux
        version: Red Hat Enterprise Linux Server 6.7 Santiago
```

**Comment by Ch3LL** [comment]
I'm able to confirm this behavior. Here is a docker container for anyone that wants to quickly replicate this.'

1. `docker run -it -v /home/ch3ll/git/salt/:/testing/ ch3ll/issues:39445 /bin/bash -c "salt-call --local state.sls test -ldebug; ls -la /dst/src/sub1"` (where /home/ch3ll/git/salt/ is a local cloned git repo of salt)
And you will see txt is owned by root and not ch3ll:

```
drwxr-xr-x. 2 root  root  17 Feb 16 21:04 .
drwxr-xr-x. 4 ch3ll ch3ll 85 Feb 16 21:07 ..
-rw-r--r--. 1 root  root   0 Feb 16 21:06 txt
```

We will need to get this fixed up thanks

**Comment by rallytime** [comment]
@systemtrap This should be fixed with #40030. It will be available in the 2016.11.4 release.
